lib/rules/multiline-comment-style.js

```diff
@@ -197,67 +302,68 @@ module.exports = {
                 }
             },
             "separate-lines"(commentGroup) {
-                if (!isJSDoc(commentGroup) && commentGroup[0].type === "Block") {
-                    const commentLines = getCommentLines(commentGroup);
-                    const block = commentGroup[0];
-                    const tokenAfter = sourceCode.getTokenAfter(block, { includeComments: true });
+                const [firstComment] = commentGroup;
 
-                    if (tokenAfter && block.loc.end.line === tokenAfter.loc.start.line) {
-                        return;
+                if (firstComment.type !== "Block" || isJSDocComment(commentGroup)) {
+                    return;
+                }
+
+                const commentLines = getCommentLines(commentGroup);
+                const tokenAfter = sourceCode.getTokenAfter(firstComment, { includeComments: true });
+
+                if (tokenAfter && firstComment.loc.end.line === tokenAfter.loc.start.line) {
+                    return;
+                }
+
+                context.report({
+                    loc: {
+                        start: firstComment.loc.start,
+                        end: { line: firstComment.loc.start.line, column: firstComment.loc.start.column + 2 }
+                    },
+                    messageId: "expectedLines",
+                    fix(fixer) {
+                        return fixer.replaceText(firstComment, convertToSeparateLines(firstComment, commentLines.filter(line => line.length)));
```

Comment:
I'm guessing this filter removes empty lines from the comment in autofix. I expect the below...

```js
        {
            code: `
                /*
                 * foo
                 *
                 * bar
                 */
            `,
            output: `
                // foo
                //${" "}
                // bar
            `,
            options: ["separate-lines"],
            errors: [{ messageId: "expectedLines", line: 2 }]
        },
        {
            code: `
                /*
                 * foo
                 *${" "}
                 * bar
                 */
            `,
            output: `
                // foo
                //${" "}
                // bar
            `,
            options: ["separate-lines"],
            errors: [{ messageId: "expectedLines", line: 2 }]
        },
```

Reply:
Thanks for catching this. Fixed!